climada/hazard/wildfire.py

```diff
@@ -0,0 +1,1165 @@
+                firms_modis = firms_modis.drop(firms_modis[ \
+                    firms_modis.confidence < CLEAN_THRESH].index)
+                temp = firms_modis
+                firms_viirs = firms.drop(firms[firms.instrument == 'MODIS'].index)
+                if firms_viirs.size:
+                    firms_viirs = firms_viirs.drop(firms_viirs[firms_viirs.confidence == 'l'].index)
+                    firms_viirs = firms_viirs.rename(columns={'bright_ti4':'brightness'})
+                    temp = temp.append(firms_viirs, sort=True)
+                    temp = temp.drop(columns=['bright_ti4'])
+
+            firms = temp
+            firms = firms.reset_index()
+            firms = firms.drop(columns=['index'])
+
+        firms['iter_ev'] = np.ones(len(firms), bool)
+        firms['cons_id'] = np.zeros(len(firms), int) - 1
+        firms['event_id'] = np.zeros(len(firms), int)
+        firms['clus_id'] = np.zeros(len(firms), int) - 1
+        firms['datenum'] = np.array(str_to_date(firms['acq_date'].values))
+        return firms
+
+    @staticmethod
+    def _firms_resolution(firms):
+        """ Returns resolution of satellite used in FIRMS in degrees
+
+        Parameters:
+            firms (pd.DataFrame): FIRMS data
+
+        Returns:
+            float
+        """
+        # Resolution in km of the centroids depends on the data origin.
+        if 'instrument' in firms.columns:
+            if firms['instrument'].any() == 'MODIS':
+                res_data = 1.0
+            else:
+                res_data = 0.375 # For VIIRS data
+        else:
+            res_data = RES_DATA # For undefined data origin, defined by user
+        return res_data/ONE_LAT_KM
+
+    @staticmethod
+    def _firms_centroids_creation(firms, res_data, centr_res_factor):
+        """ Get centroids from the firms dataset and refactor them.
+
+        Parameters:
+            firms (DataFrame): dataset obtained from FIRMS data
+            res_data (float): FIRMS instrument resolution in degrees
+            centr_res_factor (float): the factor applied to voluntarly decrease/increase
+                the centroids resolution
+
+        Returns:
+            centroids (Centroids)
+        """
+        centroids = Centroids()
+        centroids.set_raster_from_pnt_bounds((firms['longitude'].min(), \
+            firms['latitude'].min(), firms['longitude'].max(), \
+            firms['latitude'].max()), res=res_data/centr_res_factor)
+        centroids.set_meta_to_lat_lon()
+        centroids.set_area_approx()
+        centroids.set_on_land()
+        centroids.empty_geometry_points()
+
+        return centroids
+
+    def _firms_cons_days(self, firms):
+        """ Compute clusters of consecutive days (temporal clusters).
+            An interruption of days_thresh is necessary to be set in two
+            different temporal clusters.
+
+        Parameters:
+            firms (dataframe): dataset obtained from FIRMS data
+
+        Returns:
+            firms
```

Comment:
Can you add more description here in terms of what additional info is now contained in `firms` or what has changed? I find the code in this function quite hard to read. And / or perhaps add one or two more comments in the code, in the style of what you already did here twice. That's quite helpful. (e.g. in the first block, what is cons_id, and what does the first sectionn do, ..